solar_classes_methods.rb

```diff
@@ -0,0 +1,179 @@
+class Planet
+  attr_reader :name
+  attr_accessor :year_length, :distance_from_sun, :diameter
+
+  def initialize(planet_name)
+    @name = planet_name
+  end
+
+  def planet_summary
+    return "#{name}: \n Takes #{year_length} Earth years to orbit the sun. \n It's distance from the sun is #{distance_from_sun} km. \n #{name}\'s diameter is #{diameter} km."
+  end
+end
+
+class SolarSystem
+  attr_reader :planets, :sys_name # evaluate if this was a good choice for planets
+  # attr_accessor :planets
+
+  def initialize(planets, sys_name)
+    @planets = planets
+    @sys_name = sys_name
+    @all_summary = Array.new
+    @planet_names = Array.new
+    @planet_hash = Hash.new
+    create_all_summary
+    create_planet_hash
+    planet_names
+
+  end
+
+  def add_new_planet(new_planet)
+    @planets << new_planet
+    create_planet_hash
+    create_all_summary
+  end
+
+  def create_planet_hash # this method ties the index +1 of the planet to the instance of the planet, for selection in the menu
+    # @planet_hash = Hash[@planet_names.zip(@planets)]
+    @planets.each.with_index(1) do |planet, index|
+      @planet_hash[index] = planet
+    end
+  end
+
+  def planet_hash
+    @planet_hash
+  end
+
+  def planet_names # creates an array of planet_names maybe this should just be built in to create_planet_hash - same info can be accessed from there.
+    @planets.each do |planet|
+      @planet_names << planet.name
+    end
+  end
+
+  def listplanets
+    numbered_planets = ["0: List all summaries"]
+    @planets.each.with_index(1) do |planet, index|
+      numbered_planet = "#{index}: #{planet.name}"
+      numbered_planets << numbered_planet
+    end
+    numbered_planets = numbered_planets.join("\n")
+    return numbered_planets
+  end
+
+# ________ the following methods can be produced by looping over @planets and printing each planet's summary
+  def create_all_summary
+    @planets.each { |planet| @all_summary << planet.planet_summary }
+  end
+
+  def print_all_summary
+    @all_summary
+  end
+
+end
+
+
+# ________ methods for other options _________
+
+
+def get_planet_name
+  print "What is the name your new planet? > "
+  new_planet_name = gets.chomp.capitalize
+  return new_planet_name
+end
+
+def get_distance # might need to convert ',' to '_' for bignums
+  print "How far is this planet from the Sun in miles? > "
```

Comment:
While I love that you've broken all these out into separate methods, for the questions this might be a little too much functional decomposition. You've got a bunch of small methods here, but they all get called in the same sequence every time. It might be easier to follow if it was all one method.